examples/net_framework/CSharpExamples/DeepSpeechClient/DeepSpeech.cs

```diff
@@ -0,0 +1,259 @@
+﻿using DeepSpeechClient.Interfaces;
```

Comment:
I'm completely new to C#, so hard to get a good feedback on those. Can you explain why the `Async`-ness ? Is this a requirement from C# or a choice ?

In all bindings we kept that very simple and upto the caller to take care of threading, to make it easier ; in the end, the caller ends up making a new thread for deepspeech for example (that's how I dealt with in Java for example)

Reply:
Initially it was synchronous but I started to noticed that if the `feedContent` is called from async context (most of the libs records in async context) and if we call `FinishStream` on an ongoing inference throws an error related to protected memory error,  If we stop the recording and wait for `feedContent` to finish we got weird output.
I decided to create a concurrent queue which is thread safe to handle buffers and prevent to loss the order of the call stack, with the buffer we can call `FinishStream` with unprocessed buffers and wait for them.

I also decided to use async because I want to create the NuGet package from it, and I think it should be ready for use in synchronous and asynchronous scenarios. 
I can add the synchronous equivalent of the methods that don't take care of threading, and then warn about it in the docs. What you think?

Reply:
@carlfm01 There are multiple items in your reply, but from my understanding of the first part, it's just that you describe bogus threading use, and it's unrelated to the bindings themselves, but just bad way of using them: the whole lib should run in its own thread.

No idea what's a `NuGet` package. I'm not against async per se, it's just that I would like we keep the bindings as small and as minimal as possible ; in all other languages, the threading "complexity" is left to the user to avoid forcing anything

Reply:
@lissyx  Ok, I'll move threading stuff to the example